lms/services/assignment.py

```diff
@@ -210,35 +212,53 @@ def is_member(self, assignment: Assignment, h_userid: str) -> bool:
         )
 
     def get_assignments(
-        self, h_userid: str | None = None, course_id: int | None = None
+        self,
+        instructor_h_userid: str | None = None,
+        course_id: int | None = None,
+        h_userids: list[str] | None = None,
     ) -> Select[tuple[Assignment]]:
         """Get a query to fetch assignments.
 
-        :params: h_userid only return assignments the users is a member of.
-        :params: course_id only return assignments that belong to this course.
+        :param instructor_h_userid: return only assignments where instructor_h_userid is an instructor.
+        :param course_id: only return assignments that belong to this course.
+        :param h_userids: return only assignments where these users are members.
         """
 
-        assignments_query = select(Assignment)
+        query = select(Assignment)
+
+        if instructor_h_userid:
+            query = query.where(
+                Assignment.id.in_(
+                    select(AssignmentMembership.assignment_id)
+                    .join(User)
+                    .join(LTIRole)
+                    .where(
+                        User.h_userid == instructor_h_userid,
+                        LTIRole.scope == RoleScope.COURSE,
```

Comment:
Should there be a course scope here? Is it possible that the user is an instructor for the _assignment_ but not the whole course?

Reply:
No, LTI roles more granular level is the course, we store these at the assignment level but that's our mistake.

Reply:
For more context the other possible scopes are:

- system: LMS-wide, it doesn't apply to instructors, just admins.

- institution: meaning "somewhere else", ie a person that's a student in one course but a instructor in another will have the more concrete type as `course` and the other as institution.

Reply:
> No, LTI roles more granular level is the course, we store these at the assignment level but that's our mistake.

Apart from the migration work, are there any obstacles to changing the DB representation in future so that it matches how we conceptually understand the domain - ie. store roles as properties of the course rather than assignment.

Reply:
The migration work involves also changing the reporting system, not an impossible task but it has more unknowns.